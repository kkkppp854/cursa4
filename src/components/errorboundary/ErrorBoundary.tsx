import React from "react";

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<{ children: React.ReactNode }, State> {
  constructor(props: { children: React.ReactNode }) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error) {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    console.error("Error caught by ErrorBoundary:", error, errorInfo); 
  }
  render() {
    if (this.state.hasError) {
      return (
        <div className="text-danger m-4">
          Помилка!
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
