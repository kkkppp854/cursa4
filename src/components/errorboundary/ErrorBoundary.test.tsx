import { render } from '@testing-library/react';
import ErrorBoundary from './ErrorBoundary';
import '@testing-library/jest-dom';

describe('ErrorBoundary', () => {
  it('renders children without error', () => {
    const { asFragment, getByText } = render(
      <ErrorBoundary>
        <div>Test content</div>
      </ErrorBoundary>
    );

    expect(getByText('Test content')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });


  it('renders fallback UI on error', () => {
   

    const ProblemChild = () => {
      throw new Error('Test error');
    };

    const { asFragment, getByText } = render(
      <ErrorBoundary>
        <ProblemChild />
      </ErrorBoundary>
    );

    expect(getByText('Помилка!')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
