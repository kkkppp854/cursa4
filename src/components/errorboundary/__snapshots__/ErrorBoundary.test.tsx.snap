// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ErrorBoundary renders children without error 1`] = `
<DocumentFragment>
  <div>
    Test content
  </div>
</DocumentFragment>
`;

exports[`ErrorBoundary renders fallback UI on error 1`] = `
<DocumentFragment>
  <div
    class="text-danger m-4"
  >
    Помилка!
  </div>
</DocumentFragment>
`;
